Prose/ProseLib/Sources/App/AppScene.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  AppScene.swift
+//  Prose
+//
+//  Created by Valerian Saliou on 9/14/21.
+//
+
+import AuthenticationFeature
+import ComposableArchitecture
+import MainWindowFeature
+import SettingsFeature
+import SwiftUI
+
+public struct AppScene: Scene {
+    @NSApplicationDelegateAdaptor(AppDelegate.self) var appDelegate
+
+    private let store: Store<AppState, AppAction>
+    private var actions: ViewStore<Void, AppAction> { ViewStore(self.store.stateless) }
+
+    @SwiftUI.State var authWindow: NSWindow?
+    @SwiftUI.State var mainWindow: NSWindow?
+
+    public init(store: Store<AppState, AppAction>) {
+        self.store = store
+    }
+
+    public init() {
+        self.init(store: Store(
+            initialState: AppState(
+                route: .auth(.init())
+            ),
+            reducer: appReducer,
+            environment: AppEnvironment.live
+        ))
+    }
+
+    public var body: some Scene {
+        WithViewStore(self.store) { viewStore in
+            login()
+                .handlesExternalEvents(matching: Set(arrayLiteral: "login"))
+            main()
+                .handlesExternalEvents(matching: Set(arrayLiteral: "main"))
+                // NOTE: [Rémi Bardon] Make sure `.onChange` is called only once.
+                //       I put it here, but it could be anywhere.
+                .onChange(of: viewStore.route) { newRoute in
+                    // TODO: [Rémi Bardon] Find a way to avoid having this callback called on every change.
```

Comment:
Please no todos "Find a way sometime".

Extend AppRoute like so:

```swift
extension AppRoute {
  enum Tag {
    case auth
    case main
  }
  
  var tag: Tag {
    switch self {
    case .auth:
      return .auth
    case .main:
      return .main
    }
  }
}
```

Then scope the ViewStore to `self.store.scope(state: \.route.tag)` and observe `.onChange(of: viewStore.state)`.

Reply:
Oh, so this time I scope the `ViewStore`? I'm not sure I completely get when you'd like me to scope them or not 🙈

Reply:
True, adding and observing `tag` would suffice. Scoping here could still make sense because you have the whole AppState at your hands whereas in the AuthenticationView you only had the values of the form.

Reply:
If you say I can scope then I will 😜 As you say, especially when I say I have the whole app state in my hands.

Reply:
Scope once per screen 👻